app/src/main/java/com/example/parkingsystemjava/mvp/view/ParkingReservationView.java

```diff
@@ -23,26 +22,74 @@ public ParkingReservationView(Activity activity, ActivityReservationBinding bind
     }
 
     @Override
-    public void showDatePicker(ListenerDateTime listenerDateTime) {
-        DateReservationDialogFragment entryReservationDialogFragment = DateReservationDialogFragment.newInstance(listenerDateTime);
+    public void showDatePicker(ListenerDateTime listenerDateTime, boolean dateSelector) {
+        DateReservationDialogFragment entryReservationDialogFragment = DateReservationDialogFragment.newInstance(listenerDateTime, dateSelector);
         FragmentManager fragmentManager = getFragmentManager();
-        entryReservationDialogFragment.show(fragmentManager, DIALOG_FRAGMENT_ENTRY_DATE_PICKER);
+        entryReservationDialogFragment.show(fragmentManager, Constants.DIALOG_FRAGMENT_ENTRY_DATE_PICKER);
+    }
+
+    @Override
+    public void showEntryDateSelected(String date) {
+        binding.editTextReservationActivityEntry.setText(date);
+    }
+
+    @Override
+    public void showExitDateSelected(String date) {
+        binding.editTextReservationActivityExit.setText(date);
+    }
+
+    @Override
+    public void showOverlapMessage() {
+        Context context = getContext();
```

Comment:
instead of repeat this code everywhere you can call a private method and just pass the string id as parameter

Reply:
done